tools/wheels/cibw_before_build_android.sh

```diff
@@ -0,0 +1,451 @@
+cpp_args = ['-DHAVE_SIN=1', '-DHAVE_COS=1', '-DHAVE_TAN=1', '-DHAVE_SINH=1', '-DHAVE_COSH=1', '-DHAVE_TANH=1', '-DHAVE_ASIN=1', '-DHAVE_ACOS=1', '-DHAVE_ATAN=1', '-DHAVE_ATAN2=1', '-DHAVE_EXP=1', '-DHAVE_LOG=1', '-DHAVE_LOG10=1', '-DHAVE_SQRT=1', '-DHAVE_FLOOR=1', '-DHAVE_CEIL=1', '-DHAVE_FABS=1', '-DHAVE_POW=1', '-DHAVE_FMOD=1', '-DHAVE_LDEXP=1', '-DHAVE_FREXP=1', '-DHAVE_MODF=1', '-DHAVE_COPYSIGN=1', '-DHAVE_FINITE=1', '-DHAVE_ISINF=1', '-DHAVE_ISNAN=1', '-DHAVE_ISFINITE=1', '-DHAVE_EXPM1=1', '-DHAVE_LOG1P=1', '-DHAVE_ASINH=1', '-DHAVE_ACOSH=1', '-DHAVE_ATANH=1', '-DHAVE_RINT=1', '-DHAVE_TRUNC=1', '-DHAVE_EXP2=1', '-DHAVE_LOG2=1', '-DHAVE_HYPOT=1', '-DHAVE_CBRT=1', '-DHAVE_NEXTAFTER=1', '-DHAVE_CSIN=1', '-DHAVE_CCOS=1', '-DHAVE_CTAN=1', '-DHAVE_CSINH=1', '-DHAVE_CCOSH=1', '-DHAVE_CTANH=1', '-DHAVE_CASIN=1', '-DHAVE_CACOS=1', '-DHAVE_CATAN=1', '-DHAVE_CASINH=1', '-DHAVE_CACOSH=1', '-DHAVE_CATANH=1', '-DHAVE_CEXP=1', '-DHAVE_CLOG=1', '-DHAVE_CPOW=1', '-DHAVE_CSQRT=1', '-DHAVE_CABS=1', '-DHAVE_CARG=1', '-DHAVE_CIMAG=1', '-DHAVE_CREAL=1', '-DHAVE_CONJ=1', '-DHAVE_STRTOLL=1', '-DHAVE_STRTOULL=1', '-DHAVE_STRTOLD=1']
+EOF
+
+echo "Generated cross file: $CROSS_FILE"
+echo "Cross file contents:"
+cat "$CROSS_FILE"
+
+# Create a symlink with a predictable name for the config
+ln -sf "$CROSS_FILE" "/tmp/android_current.meson.cross"
+
+echo "--- Android build environment configured ---"
+echo "Using cibuildwheel's built-in Android toolchain"
+echo "Cross file: /tmp/android_current.meson.cross"
+
+# Debug Python configuration to understand the BLDLIBRARY issue
+echo "--- Debugging Python configuration ---"
+python -c "
+import sysconfig
+import os
+
+# Check BLDLIBRARY configuration
+bldlibrary = sysconfig.get_config_var('BLDLIBRARY')
+print('BLDLIBRARY from sysconfig:', repr(bldlibrary))
+
+# Check if it's in environment
+bldlib_env = os.environ.get('BLDLIBRARY', 'NOT_SET')
+print('BLDLIBRARY from env:', repr(bldlib_env))
+
+# Get other Python config vars that might be relevant
+ldlibrary = sysconfig.get_config_var('LDLIBRARY') 
+print('LDLIBRARY:', repr(ldlibrary))
+
+libs = sysconfig.get_config_var('LIBS')
+print('LIBS:', repr(libs))
+
+# Check library directory
+libdir = sysconfig.get_config_var('LIBDIR')
+print('LIBDIR:', repr(libdir))
+"
+
+# Try different approaches to fix BLDLIBRARY issue
+# The issue is meson-python is inserting literal '$(BLDLIBRARY)' in link commands
+# For Android, we don't want to link Python extensions to libpython at all
```

Comment:
We absolutely do want to link them to libpython, but if it's using the Python .pc file to achieve this, then this may be related to https://github.com/python/cpython/issues/138800.